hibernate-core/src/main/java/org/hibernate/internal/util/StringHelper.java

```diff
@@ -789,6 +789,12 @@ public static StringBuilder buildBatchFetchRestrictionFragment(
 		}
 	}
 
+	public static StringBuilder buildBatchFetchArrayRestrictionFragment(
+			String alias,
+			String columnName) {
```

Comment:
Currently, no other DB support that. We could do what you say when we have at least 2 DBs supporting this feature.

Reply:
I don't think that's a very good approach. Let's try to at least do things in a way that it could be used by other DBs at some point.

Here you are introducing a PostgreSQL specific thing in general code. That's really not something we want to do.